README.md

```diff
@@ -91,113 +94,223 @@ The front end of the integration is now complete.
+
+    return YES;
+}
+```
+then in your viewController class inside your button IBAction function call the function  : 
+
+```swift
+
+public static func startScenario(for useCaseID: String, with delegate: YotiSDKDelegate) throws 
+```
+by passing it the useCaseID and self as delegate like this:
+
+Swift:
+
+```swift
+
+@IBAction func yotiButtonDidTouchUpInside(_ sender: YotiButton) {
+	guard let useCaseID = sender.useCaseID else{
+		return
+	}
+	do {
+		UIApplication.shared.isNetworkActivityIndicatorVisible = true
+		try YotiSDK.startScenario(for: useCaseID, with: self)
+    } catch {
+		// Handle error here
+	}
+}
+```
+
+Objective-C: 
+
+```objective-c
+
+- (IBAction)buttonDidTouchUpInside:(YotiButton*)sender {
+	NSString* useCaseID = sender.useCaseID;
+	NSError* error = nil;
+
+	if (![useCaseID isEqual:@""]) {
+		[YotiSDK startScenarioForUseCaseID:useCaseID withDelegate:self error:&error];
+
+		if (error != nil) {
+			NSLog(@"error : %@", error.description);
+		}
+	}
+}
+```
+
+In Swift your ViewController class should comply to YotiSDKDelegate and to BackendDelegate in order to get the callbacks.
+
+```swift
+
+extension ViewController: YotiSDKDelegate {
+    func yotiSDKDidFail(for useCaseID: String, with error: Error) {
+    	//handle here the error opening of Yoti app
+    }
+
+    func yotiSDKDidSucceed(for useCaseID: String, baseURL: URL?, token: String?, url: URL?) {
+        //Handle here the success of the opening of Yoti app for example by requesting a profile from the backend like below
+        //Get the specific scenario by calling  
+        let scenario = YotiSDK.scenario(for: useCaseID)
+        //request the backend to get the profile linked to a specific scenario by passing the token returned and self as delegate for a call back
+        YotiSDK.callbackBackend(scenario: scenario!, token: token!, with: self)
+    }
+
+    func yotiSDKDidOpenYotiApp() {
+    //Handle specific behaviour if needed when the Yoti App didOpen
+
+    }
+}
+```
+
+In Objective-C, your viewController should comply to  YTBSDKDelegate and YTBBackendDelegate like this:
+
+```objective-c
+
+@interface ViewController () <YTBSDKDelegate, YTBBackendDelegate>
+```
+
+then implement the delegate functions of the protocol it complies to like this:
+
+```objective-c
+
+- (void)yotiSDKDidFailFor:(NSString * _Nonnull)useCaseID with:(NSError * _Nonnull)error {
+  // handle failure here
+}
+
+- (void)yotiSDKDidSucceedFor:(NSString * _Nonnull)useCaseID baseURL:(NSURL * _Nullable)baseURL token:(NSString * _Nullable)token url:(NSURL * _Nullable)url {
+	YTBScenario *scenario = [YotiSDK scenarioforUseCaseID:useCaseID];
+	[YotiSDK callbackBackendScenario:scenario token:token withDelegate:self];
+}
+
+- (void)yotiSDKDidOpenYotiApp {
+	//behaviour when SDK opens yoti app (if needed)
+}
 ```
-And then you can add the scenario to Yoti SDK with
+when the callback returns from the backend we get the data linked to the profile or the error in
```

Comment:
You are still missing documentation on the optimality of YTBBackendDelegate (and its name in swift)

Reply:
@CharlesVu : Can you please be more specific here. I didn't understand this comment. Thanks.